eidas_node/tests/test_saml.py

```diff
@@ -0,0 +1,259 @@
+    LIGHT_RESPONSE_DICT = OrderedDict(
+        [('id', '_751e557772344aa59e9e3f35d2c9f6d6'),
+         ('in_response_to_id', 'e399fb9b-9454-4284-831f-4aa33d83757e'),
+         ('issuer', 'urn:microsoft:cgg2010:fpsts'),
+         ('ip_address', '217.31.205.1'),
+         ('relay_state', 'relay123'),
+         ('subject', 'CZ/CZ/ff70c9dd-6a05-4068-aaa2-b57be4f328e9'),
+         ('subject_name_id_format', NameIdFormat.PERSISTENT),
+         ('level_of_assurance', LevelOfAssurance.SUBSTANTIAL),
+         ('status', OrderedDict(
+             [('failure', False),
+              ('status_code', StatusCode.SUCCESS),
+              ('sub_status_code', None),
+              ('status_message', None)])),
+         ('attributes', OrderedDict([
+             ('http://eidas.europa.eu/attributes/naturalperson/CurrentFamilyName', ['ČERNÝCH']),
+             ('http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName', ['PAVEL']),
+             ('http://eidas.europa.eu/attributes/naturalperson/DateOfBirth', ['1956-07-15']),
+             ('http://eidas.europa.eu/attributes/naturalperson/PlaceOfBirth', ['Praha 2']),
+             ('http://eidas.europa.eu/attributes/naturalperson/CurrentAddress',
+              ['PGVpZGFzOkxvY2F0b3JEZXNpZ25hdG9yPjEwPC9laWRhczpMb2NhdG9yRGVzaWduYXRvcj4NCjxlaWRhczpUaG9yb3VnaGZhcm'
+               'U+WmEgcGlsb3U8L2VpZGFzOlRob3JvdWdoZmFyZT4NCjxlaWRhczpQb3N0TmFtZT7EjGVza8OhIEthbWVuaWNlPC9laWRhczpQ'
+               'b3N0TmFtZT4NCjxlaWRhczpQb3N0Q29kZT40MDcyMTwvZWlkYXM6UG9zdENvZGU+DQo8ZWlkYXM6Q3ZhZGRyZXNzQXJlYT7EjG'
+               'Vza8OhIEthbWVuaWNlLCBEb2xuw60gS2FtZW5pY2U8L2VpZGFzOkN2YWRkcmVzc0FyZWE+DQo=']),
+             ('http://eidas.europa.eu/attributes/naturalperson/PersonIdentifier',
+              ['CZ/CZ/ff70c9dd-6a05-4068-aaa2-b57be4f328e9'])]))
+         ])
+    FAILED_LIGHT_RESPONSE_DICT = OrderedDict(
+        [('id', '_751e557772344aa59e9e3f35d2c9f6d6'),
+         ('in_response_to_id', 'e399fb9b-9454-4284-831f-4aa33d83757e'),
+         ('issuer', None),
+         ('ip_address', None),
+         ('relay_state', 'relay123'),
+         ('subject', 'unknown'),
+         ('subject_name_id_format', NameIdFormat.UNSPECIFIED),
+         ('level_of_assurance', LevelOfAssurance.LOW),
+         ('status', OrderedDict(
+             [('failure', True),
+              ('status_code', StatusCode.REQUESTER),
+              ('sub_status_code', SubStatusCode.REQUEST_DENIED),
+              ('status_message', 'Something went wrong.')])),
+         ('attributes', {})
+         ])  # type: Dict[str, Any]
+
+    def test_create_light_response_not_encrypted(self):
+        self.maxDiff = None
+        with cast(BinaryIO, (DATA_DIR / 'saml_response.xml').open('rb')) as f:
+            saml_response = SAMLResponse(parse_xml(f), 'relay123')
+
+        light_response = saml_response.create_light_response()
+        self.assertDictEqual(light_response.get_data_as_dict(), self.LIGHT_RESPONSE_DICT)
```

Comment:
This shouldn't be necessary, the `assertEqual` should be smart enough.

Reply:
In the case of test failure, `assertDictEqual` shows a much nicer description of the failure than plain `assertEqual`.